tasks/docs.ts

```diff
@@ -1,128 +1,37 @@
-import { Dgeni } from "dgeni";
 import gulp from "gulp";
-import htmlReplace from "gulp-html-replace";
-import htmlMin from "gulp-htmlmin";
-import imageMin from "gulp-imagemin";
-import replace from "gulp-replace";
-import del = require("del");
-import webpack = require("webpack");
+import DocsEngine from "carbonldp-ts-docs-engine";
+import path from "path";
 
+const projectRootDir:string = path.resolve( __dirname, "../" );
+const sourceDir:string = path.resolve( projectRootDir, "src/" );
+const outputDir:string = path.resolve( projectRootDir, "docs/" );
 
-export const docsCleanHTML:gulp.TaskFunction = () =>
-	del( [
-		"docs/**", "!docs",
-		"!docs/styles", "!docs/styles/**",
-		"!docs/fonts", "!docs/fonts/**",
-		"!docs/images", "!docs/images/**",
-		"!docs/scripts", "!docs/scripts/**",
-		"!docs/scripts", "!docs/scripts/**",
-		"!docs/favicon.png",
-	] );
-docsCleanHTML.displayName = "docs:clean-html";
+export const docsBuildProd:gulp.TaskFunction = () => {
 
-export const docsCreateHTML:gulp.TaskFunction = () => {
-	const dgeni = new Dgeni( [ require( "../build/docs/dgeni" ) ] );
-	return dgeni.generate();
-};
-docsCreateHTML.displayName = "docs:create-html";
-
-export const docsHTMLReplace:gulp.TaskFunction = () =>
-	gulp.src( "docs/**/*.html" )
-		.pipe( htmlReplace( {
-			"css": "assets/styles.min.css",
-			"js": "scripts/bundle.min.js"
-		} ) )
-		.pipe( replace( /<(?!base)([^>]*?)href="(\/)([^*]*?)"([?>]*?)>/gm, `<$1href="$3"$4>` ) )
-		.pipe( gulp.dest( "docs/" ) );
-docsHTMLReplace.displayName = "docs:html-replace";
-
-export const docsHTMLMin:gulp.TaskFunction = () =>
-	gulp.src( "docs/**/*.html" )
-		.pipe( htmlMin( {
-			collapseWhitespace: true,
-			removeComments: true
-		} ) )
-		.pipe( gulp.dest( "docs" ) );
-docsHTMLMin.displayName = "docs:html-min";
-
-export const docsHTML:gulp.TaskFunction = gulp.series(
-	docsCleanHTML,
-	docsCreateHTML,
-);
-docsHTML.displayName = "docs:html";
-
-
-export const docsCleanBundle:gulp.TaskFunction = () =>
-	del( [
-		"docs/styles/**",
-		"docs/scripts/**",
-		"docs/fonts/**",
-	] );
-docsCleanBundle.displayName = "docs:clean-bundle";
-
-export const docsCreateBundle:gulp.TaskFunction = done => {
-	let config = process.env.NODE_ENV === "prod" ?
-		require( "../build/docs/webpack/webpack.prod" ) :
-		require( "../build/docs/webpack/webpack.dev" );
-
-	const compiler = webpack( config );
+	let options:DocsEngine.Options = {
+		src: sourceDir,
+		out: outputDir,
+		mode: "production" as "production",
+		logLevel: "info" as "info",
```

Comment:
The casting is not necessary, since using the DocsEngine.Options types would do it automatically.